src/content_cao.h

```diff
@@ -124,7 +124,10 @@ class GenericCAO : public ClientActiveObject
 	{
 		return ACTIVEOBJECT_TYPE_GENERIC;
 	}
-
+	inline const ItemGroupList &getGroups()
```

Comment:
Since this is a getter you can also make the whole function const, as it doesn't modify the class object.